### PR TITLE
[MI-2547]:Fixed review fixes for Github issue 613 and PR 626

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-github/pull/626

### CODE REVIEW

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/client/client.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ export default class Client {
         return this.doGet(`${this.url}/connected?reminder=${reminder}`);
     }
 
-    getSidebarContent = async () => this.doGet(`${this.url}/sidebar-content`);
+    getSidebarContent = async () => {
+        return this.doGet(`${this.url}/sidebar-content`);
+    }
```
Comment (avas27JTG) [inline webapp/src/client/client.js:20]: why this? isn't it working without `{ return </> }`

File: webapp/src/websocket/index.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import {
 import {id as pluginId} from '../manifest';
 
 let timeoutId;
+const RECONNECT_JITTER_MAX_TIME = 10;
```
Comment (avas27JTG) [inline webapp/src/websocket/index.js]: mention unit of time in variable name

File: webapp/src/components/sidebar_right/index.jsx [hunk 3]
```diff
@@ -5,46 +5,22 @@ import {connect} from 'react-redux';
...
-            mergeable: foundDetails.mergeable,
-            requestedReviewers: foundDetails.requestedReviewers,
-            reviews: foundDetails.reviews,
-        };
-    });
-}
+import SidebarRight from './sidebar_right.jsx';
 
 function mapStateToProps(state) {
+    const sidebarData = getSidebarData(state);
```
Comment (avas27JTG) [inline webapp/src/components/sidebar_right/index.jsx]: destructure the needed fields directly here like `const {username, ...} = getSidebarData(state)`

File: webapp/src/components/sidebar_right/sidebar_right.jsx [hunk 4]
```diff
@@ -111,27 +111,27 @@ export default class SidebarRight extends React.PureComponent {
         switch (this.props.rhsState) {
         case RHSStates.PRS:
 
-            githubItems = this.props.yourPrs || [];
+            githubItems = this.props.yourPrs;
```
Comment (avas27JTG) [inline webapp/src/components/sidebar_right/sidebar_right.jsx]: you can destructure the props

File: webapp/src/selectors.js [hunk 5]
```diff
@@ -19,4 +23,45 @@ export const getServerRoute = (state) => {
...
+    getPluginState,
+    (pluginState) => {
+        return {
+            username: pluginState.username,
+            reviews: mapPrsToDetails(pluginState.sidebarContent.reviews || emptyArray, pluginState.reviewDetails),
+            yourPrs: mapPrsToDetails(pluginState.sidebarContent.prs || emptyArray, pluginState.yourPrDetails),
+            yourAssignments: pluginState.sidebarContent.assignments || emptyArray,
+            unreads: pluginState.sidebarContent.unreads || emptyArray,
+            org: pluginState.organization,
+            rhsState: pluginState.rhsState,
```
Comment (avas27JTG) [inline webapp/src/selectors.js]: can destructure `pluginState`

File: webapp/src/client/client.js (hunk 1 above)
Comment (Kshitij-Katiyar) [reply webapp/src/client/client.js:20]: This was one of the review comments by mattermost team to make this consistent for the whole file

File: webapp/src/websocket/index.js (hunk 2 above)
Comment (Kshitij-Katiyar) [reply webapp/src/websocket/index.js]: This name is given by mattermost team and I copied this from there

Comment (avas27JTG) [reply webapp/src/websocket/index.js]: then you can add comment for unit

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**